Sources/Bond/AppKit/NSOutlineView+Changeset.swift

```diff
@@ -27,22 +27,21 @@
 import AppKit
 import ReactiveKit
 
-open class OutlineViewBinder<Changeset: TreeChangesetProtocol>: NSObject, NSOutlineViewDataSource where Changeset.Collection: TreeArrayProtocol {
+open class OutlineViewBinder<Changeset: TreeChangesetProtocol>: NSObject, NSOutlineViewDataSource {
 
-    // NSOutlineView display a tree data structure that does not have a single root,
-    // thus we need to bind TreeArray instead of TreeNode. TreeArray is an array of TreeNodes that implements TreeNodeProtocol.
+    // NSOutlineView display a tree data structure that does not have a single root.
     //
     // "Items" of type `Any` that NSOutlineView asks for in the data source methods are tree nodes themselves, in our case `TreeNode`.
     // Note that we can't use value types here because NSOutlineView is a ObjC type so we need to use TreeArray.Object and TreeNode.Object variants.
-    //
-    // What NSOutlineView calls "object value of an item" is the value associated with a tree node: `node.value`.
-
-    public typealias IsItemExpandable = (Changeset.Collection.ChildNode, NSOutlineView) -> Bool
+    
+    public typealias IsItemExpandable = (Changeset.Collection.Children.Element, NSOutlineView) -> Bool
+    public typealias ObjectValueForItem = (Changeset.Collection.Children.Element) -> AnyObject?
```

Comment:
Is there any limitation that would prevent this from being `Any?` instead of `AnyObject?`? Seems like it might be _really_ nice to pass around raw Swift structs and enum values?

Reply:
Looks like the outline view actually accepts `Any?` so I've changed this to `Any?` as well. That being said, the delegate method is called `...objectValueFor...` so it's hard to say if all Swift types will work. My guess would be that only types that can be exposed to ObjC will work like bridgeable struct types such as String or Int based enums.

Reply:
Thankfully, I've tested this and it seems to work just fine with enums as well. Let me check the specifics, but it seems to work just fine.